docx/text/tabstops.py

```diff
@@ -18,3 +18,16 @@ class TabStops(ElementProxy):
     using the `tab_stops` property of ParagraphFormat; it is not intended to
     be constructed directly.
     """
+
+    __slots__ = ()
+
+    def __len__(self):
+        tabs = self._get_tabs()
+        return len(tabs.tab_lst)
+
+    def _get_tabs(self):
+        """
+        Get or Create tabs list as needed.
+        """
+        tabs = self._element.get_or_add_tabs()
+        return tabs
```

Comment:
I've gone back and forth on that over the course of developing the library, whether to put the collection in the same file as the item it contains or whether to put the collection in the file of the object that "owns" the collection. I've pretty much settled on putting them both in the same file, so you'll see that pattern most places in the library. `docx.section` is an example of the former. `docx.style.styles` is a counter-example.

Anyway, in this case, let's locate them both in the same file until there's some compelling reason for them to be separate. I'm expecting the total code in that module to come in well under 1000 lines (which is an informal limit for this library) and I expect TabStops by itself would be pretty short.